packages/dx-grid-core/src/plugins/table-group-header/helpers.js

```diff
@@ -0,0 +1,85 @@
+import { TABLE_BAND_TYPE } from './constants';
+
+export const isBandedTableRow = tableRow => (tableRow.type === TABLE_BAND_TYPE);
+export const isBandedOrHeaderRow = tableRow =>
+  isBandedTableRow(tableRow) || tableRow.type === 'heading';
+
+export const getColumnMeta = (columnName, columnGroups, tableRowLevel) => {
+  let currentBandTitle = null;
+  let columnLevel = 0;
+
+  const treeProcessing = (bands, level = 0, title) => {
```

Comment:
Can we change this function not to modify external variables but to return a value instead?